File: ExamenMoria-Python/projecto/models/items/Carcaj.py
from Item import Item
from interfaz.Elfo import Elfo

class Carcaj(Item, Elfo):
    def __init__(self, id: int, flechasIniciales: int):
        super().__init__(id)
        self.flechasTotales = flechasIniciales

    def __str__(self) -> str:
        return f"Carcaj({self.id}, {self.flechasTotales})"

    def lanzarFlecha(self) -> bool:
        if self.flechasTotales == 0:
            print("No quedan más flechas para disparar a los enemigos.")
            print()
            return False
        self.flechasTotales -= 1
        print("Has disparado una flecha matando a un enemigo, te quedán "+str(self.flechasTotales)+" flechas en el carcaj.")
        print()
        return True

    def recargarCarcaj(self, flechas: int):
        self.flechasTotales += flechas
        print("Se han encontrado "+str(self.flechasTotales)+" flechas en la sala y se las han quedado, ahora disponemos de "+str(flechas)+" felchas en el carcaj.")

File: ExamenMoria_Java/src/main/java/models/item/Carcaj.java
package models.item;

import interfaces.Elfo;

import java.util.UUID;

public class Carcaj extends Item implements Elfo {
    public int flechas;

    public Carcaj(UUID id, int flechas) {
        super(id);
        this.flechas = flechas;
    }

    @Override
    public String toString() {
        return "Carcaj{" +
                "flechas=" + flechas +
                ", id=" + id +
                '}';
    }

    /**
     * función que permite lanzar flechas
     * @return true si tiene para lanzar flechas y false si no puede tirar flechas, aprovechar para utilizar un break
     */
    @Override
    public boolean lanzarFlecha() {
        if(flechas == 0){
            System.out.println("No quedan más flechas para disparar a los enemigos.");
            System.out.println();
            return false;
        }
        flechas--;
        System.out.println("Has disparado una flecha matando a un enemigo, te quedán "+flechas+" flechas en el carcaj.");
        System.out.println();
        return true;
    }

    @Override
    public void recargarCarcaj(int flechas) {
        this.flechas += flechas;
        System.out.println("Se han encontrado "+flechas+" flechas en la sala y se las han quedado, ahora disponemos de "+this.flechas+" felchas en el carcaj.");
        System.out.println();
    }
}
